### ISSUE TITLE
Field resolver is called without providing resolver class to `resolvers` property in `buildSchema`

### ISSUE THREAD

**Issue by rfgamaral** [MichalLytek/type-graphql#584]
I have this `LaunchResolver` defined like this:

```ts
@Resolver(() => Launch)
class LaunchResolver implements ResolverInterface<Launch> {
    // (...)
}
```

The `Launch` type is something like this:

```ts
@ObjectType()
class Launch {
    // (other fields)

    @Field(() => Mission)
    mission!: Mission;
}
```

And then I added a `MissionResolver` with a `@FieldResolver`:

```ts
enum PatchSize {
    LARGE,
    SMALL,
}

registerEnumType(PatchSize, {
    name: 'PatchSize',
});

@Resolver(() => Mission)
class MissionResolver implements ResolverInterface<Mission> {
    @FieldResolver()
    patchUrl(
        @Root() mission: Mission,
        @Arg('size', () => PatchSize, { nullable: true }) size = PatchSize.SMALL
    ): string | undefined {
        return size === PatchSize.SMALL ? mission.smallPatchUrl : mission.largePatchUrl;
    }
}
```

And, for reference (in case it's relevant), this is the `Mission` type:

```ts
@ObjectType()
class Mission {
    @Field({ nullable: true })
    name?: string;

    @Field({ nullable: true })
    patchUrl?: string;

    smallPatchUrl?: string;

    largePatchUrl?: string;
}
```

Everything was working perfectly with the code above and it was when I realized that my `schema` was being defined like this:

```ts
const schema = buildSchemaSync({
    container: ({ context }: ResolverData<GraphQLContext>) => {
        return context.container;
    },
    resolvers: [LaunchResolver],
});
```

As you can see, `MissionResolver` is missing from the `resolvers` array but everything still works perfectly and `patchUrl` is successfully resolved with the `@FieldResolver`. I tested adding `MissionResolver` to the `resolvers` array and it worked just the same.

My question is two-fold:

1. Why is my code working without adding `MissionResolver` to the `resolvers` array?
2. What's the recommended best practice for this? Should I always add **all** resolvers or just the "main" one?

**Comment by MichalLytek** [comment]
> Why is my code working without adding MissionResolver to the resolvers array?

See #234 👀 

> What's the recommended best practice for this? 

Update to `type-graphql@beta` 😉

**Comment by rfgamaral** [comment]
> > Why is my code working without adding MissionResolver to the resolvers array?
> 
> See #234 👀

Based on my basic understanding of what I read, it feels like a somewhat different issue but maybe related, couldn't tell.


> > What's the recommended best practice for this?
> 
> Update to `type-graphql@beta` 😉

Updated to `0.18.0-beta.12` but I didn't notice any difference. Everything still worked as I expected without adding `MissionResolver` to the `resolvers` array so my quesion kinda persists... Is it supposed to?

**Comment by MichalLytek** [comment]
Oh, indeed, field resolvers are not filtered by `resolvers` array, like queries or mutations, but should be 😱 

>  Everything still worked as I expected without adding MissionResolver to the resolvers array 

Because metadata storage is global to collect metadata from each decorator, so this is the leak because of no proper filtering 😉

**Comment by MichalLytek** [comment]
Fixed via 86d1fea 🔒

**Comment by rfgamaral** [comment]
Awesome, thanks 👍

Nós eagerly waiting for the next beta release ☺️

**Comment by MichalLytek** [comment]
@rfgamaral 
Released as `v0.18.0-beta.14` 💪

**Comment by rfgamaral** [comment]
I just tested this. It worked great, thanks!
